newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -258,20 +258,49 @@ object FingerpostWireEntry
         )
     }
 
-    // grr annoying but broadly I think subjects and keywords are the same "axis" to search on
-    val keywordsOrSubjectsQuery = (keywordsQuery, subjectsQuery) match {
-      case (Some(kwq), Some(subq)) => Some(sqls"$kwq OR $subq")
-      case _                       => keywordsQuery orElse subjectsQuery
+    val categoryCodesInclQuery = search.categoryCodesIncl match {
+      case Nil => None
+      case categoryCodes =>
+        Some(
+          sqls"${syn.categoryCodes} && array[${categoryCodes.map(code => sqls"$code")}]::text[]"
+        )
+    }
+
+    val categoryCodesExclQuery = search.categoryCodesExcl match {
+      case Nil => None
+      case categoryCodesExcl =>
+        val cce = this.syntax("categoryCodesExcl")
```

Comment:
i think it would be more readable if this was inline myself, rather than single use variable with short name

Reply:
I see your point, although `cce` is actually reused later in the block. I don't understand the query lib very well so am just following the existing pattern in this file tbh. A refactor could be worthwhile (though we should probably check with Andrew, who knows the library better than me) but if so I feel like it'd be better done alongside refactoring the other queries?